### PR TITLE
Moves constant buffers to being global for ray tracing

### PR DESCRIPTION
**Pull request by anisunity**
This is a newever version of this PR: https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/5461

For the moment i was not able to see a performance improvement while doing this.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/GlobalShaderVariables.hlsl [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#ifndef UNITY_RAY_TRACING_GLOBAL_RESOURCES_INCLUDED
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/GlobalShaderVariables.hlsl]: In my upcoming Constant Buffer refactor, I renamed UnityGlobal to ShaderVariablesGlobal to match our other constant buffers naming. The result is that I have now two files ShaderVariablesGlobal.cs and .cs.hlsl.
We should rename this one to ShaderVariablesGlobal.hlsl to match the naming convention.

File: com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/GlobalShaderVariables.hlsl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#define UNITY_RAY_TRACING_GLOBAL_RESOURCES_INCLUDED
+
+#define DEBUG_RAY_TRACING_USE_GLOBAL_RESOURCES 1
+
+#if (SHADER_STAGE_RAY_TRACING && UNITY_RAY_TRACING_GLOBAL_RESOURCES && DEBUG_RAY_TRACING_USE_GLOBAL_RESOURCES)
+	#define RAY_TRACING_USE_GLOBAL_RESOURCES
+#endif
+
+#if defined(RAY_TRACING_USE_GLOBAL_RESOURCES)
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/GlobalShaderVariables.hlsl]: Should this be in this file?
Seems like this sort of macro definition are more of a "Core" thing?

**Comment by anisunity** [comment]
Ok i probably did not reopen after a rebase. checking this

**Comment by anisunity** [comment]
@remi-chapelain issue fixed

**Comment by anisunity** [comment]
Ok i was able to see the performance improvement. @remi-chapelain i can show you if you want to test

**Comment by anisunity** [comment]
@JulienIgnace-Unity review corrections done

**Pull request merged**